### PR TITLE
Resizing images on the media model (Issue #179)

### PR DESCRIPTION
**Pull request by annalee**
Resolves https://github.com/18F/peace-corps-product-management/issues/179

As with the SirTrevor image processor, creates and saves resized copies of any images uploaded to the Media model. Currently saving to the same place Sir Trevor saves--I'll be removing this functionality from Sir Trevor's processor once its images are being properly saved to the media model.

Question: This is currently pulling filenames for the resized images from the title field. Any preference to generate filenames a different way?

### CODE REVIEW

File: peacecorps/peacecorps/models.py [hunk 1]
```diff
@@ -210,6 +214,24 @@ class Media(models.Model):
...
+
+            for ext, width, height in SIZES:
+                with tempfile.TemporaryFile() as buffer_file:
+                    image.thumbnail((width, height), Image.ANTIALIAS)
+                    path = os.path.join(
+                        settings.RESIZED_IMAGE_UPLOAD_PATH,
+                        filename + '-' + ext + '.' + image.format.lower())
+                    image.save(buffer_file, image.format.lower())
+                    default_storage.save(path, buffer_file)
+        super(Media, self).save(*args, **kwargs)
```
Comment (cmc333333) [inline peacecorps/peacecorps/models.py:243]: This is just copy-paste from the sir trevor file, right? Why not reuse that function?

**Comment by annalee** [comment]
copy-paste with a minor change. Since Sir Trevor images will be getting saved to the Media model, it makes more sense to me to move the function here and remove it from Sir Trevor once I get the save working right (which should be later today; I've almost got that finished).

**Comment by cmc333333** [comment]
Okidok. Would you mind writing some tests for this if it's hanging around?

**Comment by annalee** [comment]
Good call. Will do.

**Comment by annalee** [comment]
I've moved the image resizing test and modified it to test image resizing on the media model. I'll update the PR for saving Sir Trevor images to the media model to remove Sir Trevor's image processor.

**Comment by annalee** [comment]
I'm merging this into the Save Sir Trevor PR (https://github.com/18F/peacecorps-site/pull/193) and closing this one, since the test changes rely on both PRs.

**Pull request closed without merging**